Kentico.Kontent.Delivery/DeliveryClient.cs

```diff
@@ -141,10 +164,18 @@ public async Task<DeliveryItemResponse> GetItemAsync(string codename, IEnumerabl
                 throw new ArgumentException("The codename of a content item is not specified.", nameof(codename));
             }
 
-            var endpointUrl = UrlBuilder.GetItemUrl(codename, parameters);
-            var response = await GetDeliverResponseAsync(endpointUrl);
+            return await GetCachedDataAsync(
+                CacheHelpers.GetItemKey(codename, parameters),
+                async () =>
+                {
+                    var endpointUrl = UrlBuilder.GetItemUrl(codename, parameters);
+                    var response = await GetDeliverResponseAsync(endpointUrl);
+
+                    return new DeliveryItemResponse(response, ModelProvider, ContentLinkResolver);
+                },
+                response => response != null,
+                CacheHelpers.GetItemDependencies);
```

Comment:
If we use decorators, we can avoid performing costly operations such as `CacheHelpers.Get*Dependencies` every time, even if the caching is not used.

Reply:
If caching is not enabled these methods are not called. It's a Func which is resolved only in CacheManager.